microprofile-metrics/README.adoc

```diff
@@ -0,0 +1,1422 @@
+
+Prometheus is a monitoring system and time series database.
+You can find more information at https://prometheus.io/. To start the Prometheus
+instance on our local computer we need:
+
+* a `prometheus.yml` file with the following content:
+
+[source,yaml]
+----
+# prometheus.yml
+global:
+  scrape_interval: 1s
+  external_labels:
+    monitor: 'my-monitor'
+
+scrape_configs:
+  - job_name: 'prometheus'
+    static_configs:
+      - targets: ['localhost:9090']
+  - job_name: 'prime-checker'
+    static_configs:
+      - targets: ['localhost:9990']
+----
+
+* Docker installed and running
+
+To start the prometheus docker container run the following command:
+
+[source]
+----
+docker run --rm --name prometheus -p 9090:9090 --network host \
+-v /path/to/prometheus.yml:/etc/prometheus/prometheus.yml:Z prom/prometheus:v2.14.0 \
+--config.file=/etc/prometheus/prometheus.yml
+----
+
+Wait for the container to start and then you can access `http://localhost:9090` in
+your browser to view the Prometheus console. Because we configured Prometheus (in
+`prometheus.yml`) to scrap our `prime-checker` service running on {productName} server
+we can directly access the collected metrics which Prometheus collects from
+`http://localhost:9990/metrics`. You can test it be typing for instance
+`application_org_quickstart_microprofile_metrics_PrimeNumberChecker_performedChecks_total`
+as the query expression and executing this query.
+
+Congratulations. Your service metrics are now queried by Prometheus every second so
+you can try to make a few request for prime numbers and see the values reflected in
+the console.
+
+Now that we have the Prometheus collecting the metrics we can expose Grafana.
+Grafana is analytics & monitoring solution which can be viewed as a nice
+presentation of metrics which we are being collected in the Prometheus. You can find
+more information about Grafana at https://grafana.com/.
+
+To run Grafana make sure that the Prometheus is started and collecting metrics. If
+you followed previous section, the prometheus should be running.
+
+To start Grafana you can run:
+
+[source]
+----
+docker run --rm -p 3000:3000 --network host grafana/grafana:6.4.4
+----
+
+This will expose Grafana console at `http://localhost:3000`. You can login with
+default admin user (`admin:admin`).
+
+The first thing we need to do with the new Grafana instance is to add a data source.
+Click the `Add data source` button on the welcome page. Select `Prometheus`. In
+the HTTP section set URL to be `http://localhost:9090` and change the `Scrape
+interval` to be `1s`. Click `Save & Test`.
+
+Now is the time to add a new dashboard with the metrics we are interested in. You
+can add a new custom dashboard and new panels manually (the UI is intuitive) or
+you can skip the manual configuration with the following simple provided dashboard:
+
+* click plus sign on the left side -> `Import`
+* paste the following JSON:
+
+[source,json]
```

Comment:
This file represents 30% of the readme lines without any benefit for the reader.
It could maybe moved to a separate file and just linked from here

Reply:
ok, good point.